### PR TITLE
Bump to Play 2.4.x compatible version

### PR DESCRIPTION
**Pull request by pedrorijo91**
Also, format code. If desired I can remove the second commit which formats the code

### CODE REVIEW

File: src/main/java/com/codacy/client/stash/client/auth/Parameter.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        return diff;
+    }
+
+    @Override
+    public String toString() {
+        return key + "=" + value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/Parameter.java]: Codacy Issue found: [The method equals() has an NPath complexity of 24](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903589/issues/source?bid=2734487&fileBranchId=3159563#l37)

File: src/main/java/com/codacy/client/stash/client/auth/Parameter.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Parameter parameter = (Parameter) o;
+
+        if (!key.equals(parameter.key)) return false;
+        if (!value.equals(parameter.value)) return false;
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/Parameter.java]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903589/issues/source?bid=2734487&fileBranchId=3159563#l44)

File: src/main/scala/com/codacy/client/stash/client/StashClient.scala [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
 import play.api.libs.ws.ning.NingWSClient
-
 import scala.concurrent.Await
 import scala.concurrent.duration._
 
 class StashClient(baseUrl: String, key: String, secretKey: String, token: String, secretToken: String) {
 
-  private lazy val KEY = ConsumerKey(key, secretKey)
-  private lazy val TOKEN = RequestToken(token, secretToken)
+  private lazy val KEY = new ConsumerKey(key, secretKey)
```
Comment (pedrorijo91) [inline src/main/scala/com/codacy/client/stash/client/StashClient.scala]: Codacy Issue found: [Field name 'KEY' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903558/issues/source?bid=2734487&fileBranchId=3159563#l14)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+                allParameters.add(param.getName(), param.getValue());
+            }
+        }
+        return allParameters.sortAndConcat();
+    }
+
+    /**
+     * Method for calculating OAuth signature using HMAC/SHA-1 method.
+     */
+    public String calculateSignature(String method, Uri uri, long oauthTimestamp, String nonce,
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903587/issues/source?bid=2734487&fileBranchId=3159563#l165)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+        String scheme = uri.getScheme();
+
+        StringBuilder sb = StringUtils.stringBuilder();
+        sb.append(scheme).append("://").append(uri.getHost());
+
+        int port = uri.getPort();
+        if (scheme.equals("http")) {
+            if (port == 80)
+                port = -1;
+        } else if (scheme.equals("https")) {
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903587/issues/source?bid=2734487&fileBranchId=3159563#l114)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+        /* 07-Oct-2010, tatu: URL may contain default port number; if so, need to extract
+         *   from base URL.
+         */
+        String scheme = uri.getScheme();
+
+        StringBuilder sb = StringUtils.stringBuilder();
+        sb.append(scheme).append("://").append(uri.getHost());
+
+        int port = uri.getPort();
+        if (scheme.equals("http")) {
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903587/issues/source?bid=2734487&fileBranchId=3159563#l111)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            Parameter parameter = (Parameter) o;
+
+            if (!key.equals(parameter.key)) return false;
+            if (!value.equals(parameter.value)) return false;
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903587/issues/source?bid=2734487&fileBranchId=3159563#l335)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+            return diff;
+        }
+
+        @Override
+        public String toString() {
+            return key + "=" + value;
+        }
+
+        @Override
+        public boolean equals(Object o) {
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [The method equals() has an NPath complexity of 24](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903587/issues/source?bid=2734487&fileBranchId=3159563#l328)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+        if (port != -1)
+            sb.append(':').append(port);
+
+        if (isNonEmpty(uri.getPath()))
+            sb.append(uri.getPath());
+
+        return sb.toString();
+    }
+
+    private String encodedParams(long oauthTimestamp, String nonce, List<Param> formParams, List<Param> queryParams) {
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [The method encodedParams() has an NPath complexity of 2250](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903587/issues/source?bid=2734487&fileBranchId=3159563#l128)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java [hunk 10]
```diff
@@ -0,0 +1,347 @@
...
+
+        StringBuilder sb = StringUtils.stringBuilder();
+        sb.append(scheme).append("://").append(uri.getHost());
+
+        int port = uri.getPort();
+        if (scheme.equals("http")) {
+            if (port == 80)
+                port = -1;
+        } else if (scheme.equals("https")) {
+            if (port == 443)
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/Codacy/stash-scala-client/file/2156903587/issues/source?bid=2734487&fileBranchId=3159563#l115)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 4 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129301/issues/source?bid=2734487&fileBranchId=3159563#l165)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 5 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129301/issues/source?bid=2734487&fileBranchId=3159563#l114)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 6 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129301/issues/source?bid=2734487&fileBranchId=3159563#l111)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 7 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129301/issues/source?bid=2734487&fileBranchId=3159563#l335)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 8 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [The method equals() has an NPath complexity of 24](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129301/issues/source?bid=2734487&fileBranchId=3159563#l328)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 9 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [The method encodedParams() has an NPath complexity of 2250](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129301/issues/source?bid=2734487&fileBranchId=3159563#l128)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 10 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129301/issues/source?bid=2734487&fileBranchId=3159563#l115)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java [hunk 11]
```diff
@@ -0,0 +1,347 @@
...
+        try {
+            signature = calculateSignature(request.getMethod(), request.getUri(), timestamp, nonce, request.getFormParams(), request.getQueryParams());
+        } catch (OAuthMessageSignerException e) {
+            e.printStackTrace();
+        }
+        String headerValue = constructAuthHeader(signature, nonce, timestamp);
+        requestBuilder.setHeader(HEADER_AUTHORIZATION, headerValue);
+    }
+
+    private String baseUrl(Uri uri) {
```
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [The method baseUrl() has an NPath complexity of 20](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129301/issues/source?bid=2734487&fileBranchId=3159563#l101)

File: src/main/scala/com/codacy/client/stash/client/StashClient.scala (hunk 3 above)
Comment (pedrorijo91) [inline src/main/scala/com/codacy/client/stash/client/StashClient.scala]: Codacy Issue found: [Field name 'KEY' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/Codacy/stash-scala-client/file/2157129294/issues/source?bid=2734487&fileBranchId=3159563#l14)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 4 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Avoid long parameter lists.](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175723/issues/source?bid=2734487&fileBranchId=3159563#l165)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 5 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175723/issues/source?bid=2734487&fileBranchId=3159563#l114)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 6 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175723/issues/source?bid=2734487&fileBranchId=3159563#l111)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 7 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [Avoid unnecessary if..then..else statements when returning booleans](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175723/issues/source?bid=2734487&fileBranchId=3159563#l335)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 8 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [The method equals() has an NPath complexity of 24](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175723/issues/source?bid=2734487&fileBranchId=3159563#l328)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 9 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [The method encodedParams() has an NPath complexity of 2250](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175723/issues/source?bid=2734487&fileBranchId=3159563#l128)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 10 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [These nested if statements could be combined](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175723/issues/source?bid=2734487&fileBranchId=3159563#l115)

File: src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java (hunk 11 above)
Comment (pedrorijo91) [inline src/main/java/com/codacy/client/stash/client/auth/OAuthSignatureCalculatorRSA.java]: Codacy Issue found: [The method baseUrl() has an NPath complexity of 20](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175723/issues/source?bid=2734487&fileBranchId=3159563#l101)

File: src/main/scala/com/codacy/client/stash/client/StashClient.scala (hunk 3 above)
Comment (pedrorijo91) [inline src/main/scala/com/codacy/client/stash/client/StashClient.scala]: Codacy Issue found: [Field name 'KEY' does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/Codacy/stash-scala-client/file/2170175716/issues/source?bid=2734487&fileBranchId=3159563#l14)

**Comment by machadoit** [comment]
Is this ok @rtfpessoa? Can we merge it?

**Pull request merged**